align/compiler/match_graph.py

```diff
@@ -274,7 +274,9 @@ def _top_to_bottom_translation(self, name, G1, Gsub, new_inst, sub_hierarchy_nam
                             for x in const.__fields_set__}
                         assert 'constraint' in sconst
                         logger.debug(f"transferred constraint instances {Gsub} from {const} to {sconst}")
-                        if len(sconst['instances']) > 0:
+                        if sconst['constraint'] == 'order' and len(sconst['instances']) > 1:
```

Comment:
@kkunal1408 , if condition is redundant.

Reply:
For order constraint: we need more than one instances.
For other we need atleast one instance.

Reply:
align/schema/constraint.py already checks this. Are you checking this again to make sure translation did not break anything? If yes, we should do this checking not only for order but also for other constraints as well (e.g. align). Else, why check here?

Reply:
align schema checks it with an assertion statement. I am checking here that during translation the constraint features remain intact otherwise align schema will throw an error.

Reply:
You are only checking for order. What if this was align? What if we have another similar constraint in the future?

Reply:
> > > @kkunal1408 , the tests below are failing:
> > > FAILED tests/pnr/test_write_constraint.py::test_group_block_hsc[high_speed_comparator_align]
> > > FAILED tests/pnr/test_write_constraint.py::test_group_block_hsc[high_speed_comparator_multiconnection]
> > > FAILED tests/pnr/test_write_constraint.py::test_group_block_hsc[high_speed_comparator_orderblock]
> > > FAILED tests/pnr/test_write_constraint.py::test_group_block_hsc[high_speed_comparator_portlocation]
> > > FAILED tests/pnr/test_write_constraint.py::test_group_block_hsc[high_speed_comparator_symmblock]
> > > FAILED tests/pnr/test_write_constraint.py::test_group_block_hsc[high_speed_comparator_symmnet]
> > > Why is there do_not_idenfity constraint in these?
> > 
> > 
> > The do_not_identify constraints are not deleted at end of annotation. We can delete them at the end of annotation. The do_not_identify impacted some hierarchies so still need to update these files.
> > I will update these files tonight, you can ignore these for now.
> 
> Let's not delete these constraints for align/pnr/checker.py to check.
> 

Can I just update the files then? For PnR these constraints are no more useful. Or do you have a better suggestion?

Reply:
hi @kkunal1408 , I believe the purpose of this test is to verify the constraint language to PnR language translation. Just to wrap up this PR, updating the files is fine. In a follow up PR, I would suggest refactoring this test to make revisions easier (e.g. test individual constraints,  compare json/objects in the test script instead of file comparison).